### PR TITLE
Sheaths, but not the rapier kind

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
and also lots of other impovements

## Changelog
:cl:
/:cl:

### CODE REVIEW

File: code/__DEFINES/~skyrat_defines/DNA.dm [hunk 1]
```diff
@@ -62,8 +62,17 @@
...
+#define MANDATORY_FEATURE_LIST list("mcolor" = "FFB","mcolor2" = "FFB","mcolor3" = "FFB","ethcolor" = "FCC","skin_color" = "FED","flavor_text" = "","breasts_size" = 1,"breasts_lactation" = FALSE,"penis_size" = 13,"penis_girth" = 9,"penis_taur_mode" = TRUE,"penis_sheath" = SHEATH_NONE ,"balls_size" = 1, "body_size" = BODY_SIZE_NORMAL)
 
 #define UNDERWEAR_HIDE_SOCKS		(1<<0)
 #define UNDERWEAR_HIDE_SHIRT		(1<<1)
 #define UNDERWEAR_HIDE_UNDIES		(1<<2)
+
+#define AROUSAL_CANT		0
+#define AROUSAL_NONE		1
+#define AROUSAL_PARTIAL		2
+#define AROUSAL_FULL		3
```
Comment (Gandalf2k15) [inline code/__DEFINES/~skyrat_defines/DNA.dm:78]: Perhaps consider renaming these to excitement?

File: modular_skyrat/modules/customization/modules/client/preferences.dm [hunk 2]
```diff
@@ -182,6 +182,8 @@
 	var/chosen_augment_slot
 	///Whether the user wants to see body size being shown in the preview
 	var/show_body_size = FALSE
+	///The arousal state of the previewed character, can be toggled by the user
+	var/arousal_preview = AROUSAL_NONE
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/customization/modules/client/preferences.dm:186]: Same here, excitement?

**Review by Gandalf2k15 — APPROVED** [review]
Consider my comments, if you want to do it it's up to you, if not, we can merge this.

File: code/__DEFINES/~skyrat_defines/DNA.dm [hunk 3]
```diff
@@ -62,8 +62,17 @@
 #define BODY_SIZE_MAX			1.25
 #define BODY_SIZE_MIN			0.85
 
-#define MANDATORY_FEATURE_LIST list("mcolor" = "FFB","mcolor2" = "FFB","mcolor3" = "FFB","ethcolor" = "FCC","skin_color" = "FED","flavor_text" = "","breasts_size" = 1,"breasts_lactation" = FALSE,"penis_size" = 13,"penis_girth" = 9,"penis_taur_mode" = TRUE,"balls_size" = 1, "body_size" = BODY_SIZE_NORMAL)
+#define SHEATH_NONE			"none"
+#define SHEATH_NORMAL		"sheath"
+#define SHEATH_SLIT			"slit"
```
Comment (Gandalf2k15) [inline code/__DEFINES/~skyrat_defines/DNA.dm]:
```suggestion
#define SHEATH_NONE			"None"
#define SHEATH_NORMAL		"Sheath"
#define SHEATH_SLIT			"Slit"
```

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]

File: code/__DEFINES/~skyrat_defines/DNA.dm (hunk 3 above)
Comment (Azarak) [reply code/__DEFINES/~skyrat_defines/DNA.dm]: They're used as text to compile the string to show a icon state, that's why they're lowercase.. hm..

Comment (Gandalf2k15) [reply code/__DEFINES/~skyrat_defines/DNA.dm]: Can you just use lowercase when getting the icon state? It looks a bit tacky having everything else as capitalized and then randomly the sheath bit completely lowercase imo

Comment (Azarak) [reply code/__DEFINES/~skyrat_defines/DNA.dm]: True, I'mma do that

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by Gandalf2k15** [comment]
All systems are go.

**Pull request merged**